### PR TITLE
Make LoopControlLoopCoalesceINTELMask parameter be mandatory

### PR DESCRIPTION
**Pull request by MrSidims**
Otherwise we can't make any assumptions about LoopControl parameters
order.

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by MrSidims** [comment]
@vmaksimo @AGindinson @mlychkov please take a look

File: lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -748,16 +748,14 @@ void SPIRVToLLVM::setLLVMLoopMetadata(const LoopInstType *LM,
...
-             "Missing loop control parameter!");
-    }
+    Metadata.push_back(llvm::MDNode::get(
+        *Context,
+        getMetadataFromNameAndParameter("llvm.loop.ivdep.safelen",
+                                        LoopControlParameters[NumParam])));
+    ++NumParam;
+    // TODO: Fix the increment/assertion logic in all of the conditions
+    assert(NumParam <= LoopControlParameters.size() &&
+           "Missing loop control parameter!");
```
Comment (AGindinson) [inline lib/SPIRV/SPIRVReader.cpp:758]: Technically, this chunk had better be factored out into a separate commit.

**Review by AGindinson — APPROVED** [review]
